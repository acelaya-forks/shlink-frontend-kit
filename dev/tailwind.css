@import 'tailwindcss' prefix(tw) important;
@import '../src/tailwind.preset.css';
