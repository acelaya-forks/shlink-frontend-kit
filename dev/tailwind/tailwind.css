@import 'tailwindcss' prefix(tw) important;
@import '../../src/tailwind/tailwind.preset.css';
