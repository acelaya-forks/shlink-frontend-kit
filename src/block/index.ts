export * from './Message';
export * from './Result';
export * from './SimpleCard';
