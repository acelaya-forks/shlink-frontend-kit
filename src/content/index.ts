export * from './Details';
export * from './Listbox';
export * from './Table';
