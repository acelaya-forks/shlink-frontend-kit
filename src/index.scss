@import './utils/ResponsiveTable';
@import './theme/theme';
