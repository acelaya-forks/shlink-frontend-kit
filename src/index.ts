export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
