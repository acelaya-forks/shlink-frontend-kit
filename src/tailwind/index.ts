export * from './content';
export * from './feedback';
export * from './form';
export * from './helpers';
export * from './navigation';
export * from './surfaces';
export * from './utils';
export * from './types';
