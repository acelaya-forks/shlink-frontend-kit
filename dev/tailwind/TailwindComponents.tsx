import type { FC } from 'react';
import { Route, Routes } from 'react-router';
import { DetailsPage } from './content/DetailsPage';
import { TablePage } from './content/TablePage';
import { MessagePage } from './feedback/MessagePage';
import { ModalDialogPage } from './feedback/ModalDialogPage';
import { ResultPage } from './feedback/ResultPage';
import { TooltipPage } from './feedback/TooltipPage';
import { ButtonsPage } from './form/ButtonsPage';
import { InputsPage } from './form/InputsPage';
import { SearchComboboxPage } from './form/SearchComboboxPage';
import { SearchInputPage } from './form/SearchInputPage';
import { NavPillsPage } from './navigation/NavPillsPage';
import { PaginatorPage } from './navigation/PaginatorPage';
import { CardsPage } from './surfaces/CardsPage';
import { UtilsPage } from './utils/UtilsPage';

export const TailwindComponents: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<h2 className="text-center">Select component - Tailwind</h2>} />
      <Route path="/form/inputs" element={<InputsPage />} />
      <Route path="/form/search-input" element={<SearchInputPage />} />
      <Route path="/form/search-combobox" element={<SearchComboboxPage />} />
      <Route path="/form/buttons" element={<ButtonsPage />} />
      <Route path="/surfaces/cards" element={<CardsPage />} />
      <Route path="/content/tables" element={<TablePage />} />
      <Route path="/content/details" element={<DetailsPage />} />
      <Route path="/navigation/paginator" element={<PaginatorPage />} />
      <Route path="/navigation/nav-pills">
        <Route path="" element={<NavPillsPage />} />
        <Route path="*" element={<NavPillsPage />} />
      </Route>
      <Route path="/feedback/dialogs" element={<ModalDialogPage />} />
      <Route path="/feedback/result" element={<ResultPage />} />
      <Route path="/feedback/message" element={<MessagePage />} />
      <Route path="/feedback/tooltip" element={<TooltipPage />} />
      <Route path="/utils" element={<UtilsPage />} />
      <Route path="*" element={<h2 className="text-center">Not found - Tailwind</h2>} />
    </Routes>
  );
};
