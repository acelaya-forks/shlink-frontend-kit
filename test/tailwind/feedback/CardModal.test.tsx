import { screen } from '@testing-library/react';
import { useState } from 'react';
import type { CardModalProps } from '../../../src/tailwind';
import { CardModal } from '../../../src/tailwind';
import { checkAccessibility } from '../../__helpers__/accessibility';
import { renderWithEvents } from '../../__helpers__/setUpTest';

describe('<CardModal />', () => {
  const onClose = vi.fn();
  const setUp = (props: Partial<CardModalProps> = {}) => renderWithEvents(
    <CardModal open onClose={onClose} title="The title" {...props}>
      <div data-testid="content">This is the content</div>
    </CardModal>,
  );

  it('passes a11y checks', () => checkAccessibility(setUp()));

  it.each([
    ['default' as const],
    ['danger' as const],
    ['cover' as const],
  ])('closes modal when close button is clicked', async (variant) => {
    const { user } = setUp({ variant });

    await user.click(screen.getByLabelText('Close dialog'));
    expect(onClose).toHaveBeenCalled();
  });

  it.each([
    { onConfirm: vi.fn() },
    { onConfirm: undefined },
  ])('shows footer only when onConfirm is provided', ({ onConfirm }) => {
    setUp({ onConfirm });

    if (onConfirm) {
      expect(screen.getByTestId('footer')).toBeInTheDocument();
    } else {
      expect(screen.queryByTestId('footer')).not.toBeInTheDocument();
    }
  });

  it.each([
    { confirmText: undefined },
    { confirmText: 'Do something' },
    { confirmText: 'Yes' },
  ])('allows confirm text to be customized', ({ confirmText }) => {
    setUp({ confirmText, onConfirm: vi.fn() });
    expect(screen.getByText(confirmText ?? 'Confirm')).toBeInTheDocument();
  });

  it.each([
    { button: 'Cancel', expectedCallback: onClose },
    { button: 'Confirm', expectedCallback: undefined },
  ])('invokes expected callback when footer buttons are clicked', async ({ button, expectedCallback }) => {
    const onConfirm = vi.fn();
    const { user } = setUp({ onConfirm });

    await user.click(screen.getByRole('button', { name: button }));

    expect(expectedCallback ?? onConfirm).toHaveBeenCalled();
  });

  it.each([
    ['sm' as const],
    ['md' as const],
    ['lg' as const],
    ['xl' as const],
  ])('renders expected size', (size) => {
    const { container } = setUp({ size });
    expect(container).toMatchSnapshot();
  });

  it.each([
    { variant: 'default' as const, onConfirm: vi.fn() },
    { variant: 'danger' as const, onConfirm: vi.fn() },
    { variant: 'cover' as const },
  ])('renders expected variant', (props) => {
    const { container } = setUp(props);
    expect(container).toMatchSnapshot();
  });

  it('defers closing the modal until the transition has finished', async () => {
    function ClosableModalWrapper() {
      const [open, setOpen] = useState(true);
      return (
        <CardModal open={open} onClose={() => setOpen(false)} title="The title">
          <div data-testid="content">This is the content</div>
        </CardModal>
      );
    }
    const { user } = renderWithEvents(<ClosableModalWrapper />);

    await user.click(screen.getByLabelText('Close dialog'));

    // Immediately after, the modal is still open
    expect(screen.getByTestId('transition-container')).toBeInTheDocument();
  });
});
