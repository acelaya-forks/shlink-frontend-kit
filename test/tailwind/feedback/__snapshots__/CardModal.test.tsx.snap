// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CardModal /> > renders expected size 1`] = `
<div>
  <dialog
    class="tw:bg-transparent tw:backdrop:bg-black/50 tw:flex tw:w-screen tw:h-screen tw:max-w-screen tw:max-h-screen"
  >
    <div
      class="tw:m-auto tw:p-4"
    >
      <div
        class="tw:group/card tw:rounded-md tw:shadow-md tw:border tw:border-lm-border tw:dark:border-dm-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:w-full tw:md:w-sm"
      >
        <div
          class="tw:px-4 tw:py-3 tw:rounded-t-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-b tw:border-lm-border tw:dark:border-dm-border tw:flex tw:items-center tw:justify-between tw:sticky tw:top-0"
        >
          <h5
            class=""
          >
            The title
          </h5>
          <button
            aria-label="Close dialog"
            class="tw:opacity-50 tw:highlight:opacity-80 tw:transition-opacity tw:rounded-md tw:focus-ring"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-xl "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="tw:p-4 tw:bg-lm-primary tw:dark:bg-dm-primary tw:first:rounded-t-md tw:first:rounded-t-md tw:last:rounded-b-md"
        >
          <div
            data-testid="content"
          >
            This is the content
          </div>
        </div>
      </div>
    </div>
  </dialog>
</div>
`;

exports[`<CardModal /> > renders expected size 2`] = `
<div>
  <dialog
    class="tw:bg-transparent tw:backdrop:bg-black/50 tw:flex tw:w-screen tw:h-screen tw:max-w-screen tw:max-h-screen"
  >
    <div
      class="tw:m-auto tw:p-4"
    >
      <div
        class="tw:group/card tw:rounded-md tw:shadow-md tw:border tw:border-lm-border tw:dark:border-dm-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:w-full tw:md:w-lg"
      >
        <div
          class="tw:px-4 tw:py-3 tw:rounded-t-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-b tw:border-lm-border tw:dark:border-dm-border tw:flex tw:items-center tw:justify-between tw:sticky tw:top-0"
        >
          <h5
            class=""
          >
            The title
          </h5>
          <button
            aria-label="Close dialog"
            class="tw:opacity-50 tw:highlight:opacity-80 tw:transition-opacity tw:rounded-md tw:focus-ring"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-xl "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="tw:p-4 tw:bg-lm-primary tw:dark:bg-dm-primary tw:first:rounded-t-md tw:first:rounded-t-md tw:last:rounded-b-md"
        >
          <div
            data-testid="content"
          >
            This is the content
          </div>
        </div>
      </div>
    </div>
  </dialog>
</div>
`;

exports[`<CardModal /> > renders expected size 3`] = `
<div>
  <dialog
    class="tw:bg-transparent tw:backdrop:bg-black/50 tw:flex tw:w-screen tw:h-screen tw:max-w-screen tw:max-h-screen"
  >
    <div
      class="tw:m-auto tw:p-4"
    >
      <div
        class="tw:group/card tw:rounded-md tw:shadow-md tw:border tw:border-lm-border tw:dark:border-dm-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:w-full tw:md:w-4xl"
      >
        <div
          class="tw:px-4 tw:py-3 tw:rounded-t-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-b tw:border-lm-border tw:dark:border-dm-border tw:flex tw:items-center tw:justify-between tw:sticky tw:top-0"
        >
          <h5
            class=""
          >
            The title
          </h5>
          <button
            aria-label="Close dialog"
            class="tw:opacity-50 tw:highlight:opacity-80 tw:transition-opacity tw:rounded-md tw:focus-ring"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-xl "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="tw:p-4 tw:bg-lm-primary tw:dark:bg-dm-primary tw:first:rounded-t-md tw:first:rounded-t-md tw:last:rounded-b-md"
        >
          <div
            data-testid="content"
          >
            This is the content
          </div>
        </div>
      </div>
    </div>
  </dialog>
</div>
`;

exports[`<CardModal /> > renders expected size 4`] = `
<div>
  <dialog
    class="tw:bg-transparent tw:backdrop:bg-black/50 tw:flex tw:w-screen tw:h-screen tw:max-w-screen tw:max-h-screen"
  >
    <div
      class="tw:m-auto tw:p-4"
    >
      <div
        class="tw:group/card tw:rounded-md tw:shadow-md tw:border tw:border-lm-border tw:dark:border-dm-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:w-full tw:md:w-6xl"
      >
        <div
          class="tw:px-4 tw:py-3 tw:rounded-t-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-b tw:border-lm-border tw:dark:border-dm-border tw:flex tw:items-center tw:justify-between tw:sticky tw:top-0"
        >
          <h5
            class=""
          >
            The title
          </h5>
          <button
            aria-label="Close dialog"
            class="tw:opacity-50 tw:highlight:opacity-80 tw:transition-opacity tw:rounded-md tw:focus-ring"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-xl "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="tw:p-4 tw:bg-lm-primary tw:dark:bg-dm-primary tw:first:rounded-t-md tw:first:rounded-t-md tw:last:rounded-b-md"
        >
          <div
            data-testid="content"
          >
            This is the content
          </div>
        </div>
      </div>
    </div>
  </dialog>
</div>
`;

exports[`<CardModal /> > renders expected variant 1`] = `
<div>
  <dialog
    class="tw:bg-transparent tw:backdrop:bg-black/50 tw:flex tw:w-screen tw:h-screen tw:max-w-screen tw:max-h-screen"
  >
    <div
      class="tw:m-auto tw:p-4"
    >
      <div
        class="tw:group/card tw:rounded-md tw:shadow-md tw:border tw:border-lm-border tw:dark:border-dm-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:w-full tw:md:w-lg"
      >
        <div
          class="tw:px-4 tw:py-3 tw:rounded-t-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-b tw:border-lm-border tw:dark:border-dm-border tw:flex tw:items-center tw:justify-between tw:sticky tw:top-0"
        >
          <h5
            class=""
          >
            The title
          </h5>
          <button
            aria-label="Close dialog"
            class="tw:opacity-50 tw:highlight:opacity-80 tw:transition-opacity tw:rounded-md tw:focus-ring"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-xl "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="tw:p-4 tw:bg-lm-primary tw:dark:bg-dm-primary tw:first:rounded-t-md tw:first:rounded-t-md tw:last:rounded-b-md"
        >
          <div
            data-testid="content"
          >
            This is the content
          </div>
        </div>
        <div
          class="tw:px-4 tw:py-3 tw:rounded-b-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-t tw:border-lm-border tw:dark:border-dm-border tw:flex tw:flex-row-reverse tw:gap-x-2 tw:items-center tw:[&]:px-3 tw:sticky tw:bottom-0"
          data-testid="footer"
        >
          <button
            class="tw:flex tw:gap-2 tw:items-center tw:justify-center tw:border tw:rounded-md tw:no-underline tw:transition-colors tw:focus-ring tw:px-3 tw:py-1.5 tw:border-brand tw:text-brand tw:text-white tw:bg-brand tw:highlight:bg-brand-dark tw:highlight:border-brand-dark tw:highlight:bg-brand"
          >
            Confirm
          </button>
          <button
            class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-3 tw:py-1.5"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </dialog>
</div>
`;

exports[`<CardModal /> > renders expected variant 2`] = `
<div>
  <dialog
    class="tw:bg-transparent tw:backdrop:bg-black/50 tw:flex tw:w-screen tw:h-screen tw:max-w-screen tw:max-h-screen"
  >
    <div
      class="tw:m-auto tw:p-4"
    >
      <div
        class="tw:group/card tw:rounded-md tw:shadow-md tw:border tw:border-lm-border tw:dark:border-dm-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:w-full tw:md:w-lg"
      >
        <div
          class="tw:px-4 tw:py-3 tw:rounded-t-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-b tw:border-lm-border tw:dark:border-dm-border tw:flex tw:items-center tw:justify-between tw:sticky tw:top-0"
        >
          <h5
            class="tw:text-danger"
          >
            The title
          </h5>
          <button
            aria-label="Close dialog"
            class="tw:opacity-50 tw:highlight:opacity-80 tw:transition-opacity tw:rounded-md tw:focus-ring"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-xl "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="tw:p-4 tw:bg-lm-primary tw:dark:bg-dm-primary tw:first:rounded-t-md tw:first:rounded-t-md tw:last:rounded-b-md"
        >
          <div
            data-testid="content"
          >
            This is the content
          </div>
        </div>
        <div
          class="tw:px-4 tw:py-3 tw:rounded-b-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:border-t tw:border-lm-border tw:dark:border-dm-border tw:flex tw:flex-row-reverse tw:gap-x-2 tw:items-center tw:[&]:px-3 tw:sticky tw:bottom-0"
          data-testid="footer"
        >
          <button
            class="tw:flex tw:gap-2 tw:items-center tw:justify-center tw:border tw:rounded-md tw:no-underline tw:transition-colors tw:focus-ring-danger tw:px-3 tw:py-1.5 tw:border-danger tw:text-white tw:bg-danger tw:highlight:bg-danger-dark tw:highlight:border-danger-dark tw:highlight:bg-danger"
          >
            Confirm
          </button>
          <button
            class="tw:inline-flex tw:rounded-md tw:focus-ring tw:text-brand tw:highlight:text-brand-dark tw:highlight:underline tw:px-3 tw:py-1.5"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </dialog>
</div>
`;
